src/Common/PathUtilities.cs

```diff
@@ -5,41 +5,27 @@ namespace Microsoft.DotNet;
 
 static class PathUtilities
 {
-    const int S_IRUSR = 256;
-    const int S_IWUSR = 128;
-    const int S_IXUSR = 64;
-    const int S_IRWXU = S_IRUSR | S_IWUSR | S_IXUSR; // 700 (octal) Permissions 
-
-    const int MAX_NUM_DIRECTORY_CREATE_RETRIES = 2;
-
     public static string CreateTempSubdirectory()
-    {
-        return CreateTempSubdirectoryRetry(0);
-    }
-
-    [DllImport("libc", SetLastError = true)]
-    private static extern int mkdir(string pathname, int mode);
-    private static string CreateTempSubdirectoryRetry(int attemptNo)
     {
         string path = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString());
-        if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
-        {
-            int mkdirStatusCode = mkdir(path, S_IRWXU);
-            if (mkdirStatusCode != 0)
-            {
-                int errno = Marshal.GetLastWin32Error();
-                if (Directory.Exists(path) && attemptNo < MAX_NUM_DIRECTORY_CREATE_RETRIES)
-                {
-                    return CreateTempSubdirectoryRetry(attemptNo + 1);
-                }
-                else
-                    throw new IOException($"Failed to create a temporary subdirectory {path} with mkdir, error code: {errno}");
-            }
-        }
+
+#if NETFRAMEWORK
+        Directory.CreateDirectory(path);
+#else
+        if (OperatingSystem.IsWindows())
+            Directory.CreateDirectory(path);
         else
         {
-            Directory.CreateDirectory(path);
+            UnixFileMode desiredMode = UnixFileMode.UserRead | UnixFileMode.UserWrite | UnixFileMode.UserExecute;
+
+            do 
+            {
+                path = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString());
```

Comment:
Thought: What if someone fills temp with every possible GUID? Then they could write a directory, write malicious content in it, then set the permissions and this code would collect it. I think at this point, /tmp/ is 'screwed' anyways (in the sense of a DDOS attack) but need to think about that.

Reply:
Great point.